src/main/java/com/thebois/models/world/World.java

```diff
@@ -210,7 +217,16 @@ private void postObstacleEvent(final int posX, final int posY) {
     }
 
     @Override
-    public Iterable<ITile> getNeighboursOf(final ITile tile) {
+    public Optional<IResource> getNearbyOfType(final Position origin, final ResourceType type) {
+        return getResources()
+            .stream()
+            .filter(resource -> resource.getType().equals(type))
+            .min((o1, o2) -> Float.compare(origin.distanceTo(o1.getPosition()),
+                                           origin.distanceTo(o2.getPosition())));
+    }
```

Comment:
Yeah, I intentionally implemented it in the most simple way (aka dumbest way) I could think of. That's because it is also the most readable, and easy to understand way. Like with the pathfinding, I chose not to preemptively optimize.

When running the game in a 50x50 world, there aren't any problems. And this isn't a method that is called every frame, it's called much more seldom than that. 

What do you think, should we be benchmarking the game with a bigger world? Or should we optimize for the 50x50 size and be happy with if that works? To be honest, I'm not sure, we might want to up the size, but then the question is where we draw the line. Any ideas?

Reply:
Good idea, where do you think mentioning that would fit in the SDD?

Reply:
Maybe known issues?

Reply:
That a good spot for it, since it is a know issue

Reply:
Added 👍